pkgs/servers/web-apps/crater/default.nix

```diff
@@ -0,0 +1,50 @@
+{ pkgs, lib, stdenv, fetchzip, nixosTests, dataDir ? "/var/lib/crater" }:
+
+stdenv.mkDerivation rec {
+  pname = "crater";
+  version = "6.0.6";
+
+  src = fetchzip {
+    url = "https://craterapp.com/downloads/file/${version}";
+    sha256 = "sha256-g+T/V+iypD6H0n8y1X7m2T6xRlYxO86aVV/yxhwHVa8=";
+    extension = "zip";
+  };
+
+  dontBuild = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    cp -ra crater $out
+
+    # symlink mutable data into the nix store due to crater path requirements
+    rm -r $out/storage $out/.env $out/bootstrap/cache $out/database/database.sqlite
+    ln -s ${dataDir}/.env $out/.env
+    ln -s ${dataDir}/storage $out/storage
+    ln -s ${dataDir}/public/storage $out/public/storage
+    ln -s ${dataDir}/bootstrap/cache $out/bootstrap/cache
+    ln -s ${dataDir}/database.sqlite $out/database/database.sqlite
+
+    runHook postInstall
+  '';
+
+  postPatch = ''
+    # Change default sqlite database path to absolute path since symlinks are
+    # not followed by crater/lavarel
+    substituteInPlace crater/app/Http/Controllers/V1/Installation/DatabaseConfigurationController.php \
```

Comment:
Looks right to me 👍

We need to cover a few other paths too though, don't we? Maybe we need to replace [this](https://github.com/crater-invoice/crater/blob/889d22d92c9cca9234d079b354c0af75be84aab2/config/filesystems.php#L48) line with something like `${cfg.dataDir}`?